api/src/main/java/org/openmrs/liquibase/ChangeLogVersionFinder.java

```diff
@@ -0,0 +1,157 @@
+	        + "core-data";
+	
+	static final String SCHEMA_ONLY_FOLDER_NAME = BASE_FOLDER_NAME + File.separator + "snapshots" + File.separator
+	        + "schema-only";
+	
+	static final String UPDATES_FOLDER_NAME = BASE_FOLDER_NAME + File.separator + "updates";
+	
+	static final String CORE_DATA_BASE_NAME = "liquibase-core-data-";
+	
+	static final String SCHEMA_ONLY_BASE_NAME = "liquibase-schema-only-";
+	
+	static final String UPDATE_TO_LATEST_BASE_NAME = "liquibase-update-to-latest-";
+	
+	private static final String DOT_XML = ".xml";
+	
+	private static final String LOWER_CASE_X = "x";
+	
+	private ChangeLogVersions changeLogVersions;
+	
+	/**
+	 * The default constructor initialises the default provider of change log versions.
+	 */
+	public ChangeLogVersionFinder() {
+		this.changeLogVersions = new ChangeLogVersions();
+	}
+	
+	/**
+	 * Allows to inject a mock provider of change log versions for unit testing.
+	 * 
+	 * @param changeLogVersions a provider of change log versions.
+	 */
+	public ChangeLogVersionFinder(ChangeLogVersions changeLogVersions) {
+		this.changeLogVersions = changeLogVersions;
+	}
+	
+	public SortedMap<String, List<String>> getChangeLogCombinations() {
+		SortedMap<String, List<String>> changeLogCombinations = new TreeMap<>();
+		
+		for (String snapshotVersion : getSnapshotVersions()) {
+			List<String> changeLogFilenames = new ArrayList<>();
+			
+			changeLogFilenames.addAll(getSnapshotFilenames(snapshotVersion));
+			
+			changeLogFilenames.addAll(getUpdateFileNames(getUpdateVersionsGreaterThan(snapshotVersion)));
+			
+			changeLogCombinations.put(snapshotVersion, changeLogFilenames);
+		}
+		
+		return changeLogCombinations;
+	}
+	
+	public SortedMap<String, List<String>> getSnapshotCombinations() {
+		SortedMap<String, List<String>> changeLogCombinations = new TreeMap<>();
+		
+		for (String snapshotVersion : getSnapshotVersions()) {
+			List<String> changeLogFilenames = new ArrayList<>();
+			
+			changeLogFilenames.addAll(getSnapshotFilenames(snapshotVersion));
+			
+			changeLogCombinations.put(snapshotVersion, changeLogFilenames);
+		}
+		
+		return changeLogCombinations;
+	}
+	
+	public List<String> getSnapshotFilenames(String version) {
+		String versionAsDotX = getVersionAsDotX(version);
+		return Arrays.asList(SCHEMA_ONLY_FOLDER_NAME + File.separator + SCHEMA_ONLY_BASE_NAME + versionAsDotX + DOT_XML,
+		    CORE_DATA_FOLDER_NAME + File.separator + CORE_DATA_BASE_NAME + versionAsDotX + DOT_XML);
+	}
+	
+	public Optional<String> getLatestSnapshotVersion() {
+		return getSnapshotVersions().stream().max(new VersionComparator());
+	}
+	
+	public Optional<String> getLatestSchemaSnapshotFilename() {
+		Optional<String> snapshotVersion = getLatestSnapshotVersion();
+		if (snapshotVersion.isPresent()) {
+			return Optional
+			        .of(SCHEMA_ONLY_FOLDER_NAME + File.separator + SCHEMA_ONLY_BASE_NAME + snapshotVersion.get() + DOT_XML);
+		}
+		return Optional.empty();
+	}
```

Comment:
This can be reduced to:

```java
return getLatestLiquibaseSnapshotVersion().map(
			snapshotVersion -> LIQUIBASE_SNAPSHOTS_FOLDER + File.separator + snapshotVersion + File.separator
				+ LIQUIBASE_CORE_DATA_FILENAME);
```

Not that you need to make that change.

Reply:
Thanks, that is more beautiful indeed.